cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go

```diff
@@ -415,17 +434,54 @@ func (m *asgCache) createPlaceholdersForDesiredNonStartedInstances(groups []*aut
 
 		klog.V(4).Infof("Instance group %s has only %d instances created while requested count is %d. "+
 			"Creating placeholder instances.", *g.AutoScalingGroupName, realInstances, desired)
+
+		healthStatus := ""
+		isUnavailable, err := m.isNodeGroupUnavailable(g)
+		if err != nil {
+			klog.V(4).Infof("Could not check instance availability, creating placeholder node anyways: %v", err)
+		} else if isUnavailable {
+			klog.Warningf("Instance group %s cannot provision any more nodes!", *g.AutoScalingGroupName)
+			healthStatus = placeholderUnfulfillableStatus
+		}
+
 		for i := realInstances; i < desired; i++ {
 			id := fmt.Sprintf("%s-%s-%d", placeholderInstanceNamePrefix, *g.AutoScalingGroupName, i)
 			g.Instances = append(g.Instances, &autoscaling.Instance{
 				InstanceId:       &id,
 				AvailabilityZone: g.AvailabilityZones[0],
+				HealthStatus:     &healthStatus,
 			})
 		}
 	}
 	return groups
 }
 
+func (m *asgCache) isNodeGroupUnavailable(group *autoscaling.Group) (bool, error) {
+	input := &autoscaling.DescribeScalingActivitiesInput{
+		AutoScalingGroupName: group.AutoScalingGroupName,
+		MaxRecords:           aws.Int64(1), // We only care about the most recent event
+	}
+
+	start := time.Now()
+	response, err := m.awsService.DescribeScalingActivities(input)
+	observeAWSRequest("DescribeScalingActivities", err, start)
+	if err != nil {
+		return false, err
+	}
+
+	if len(response.Activities) > 0 {
+		activity := response.Activities[0]
+		asgRef := AwsRef{Name: *group.AutoScalingGroupName}
+		lut := m.registeredAsgs[asgRef].lastUpdateTime
+		if lut == nil || activity.StartTime.Before(*lut) {
+			return true, nil
+		} else if *activity.StatusCode == "Failed" {
+			return false, nil
+		}
+	}
+	return true, nil
```

Comment:
should these boolean returns be flipped? Why do we return true (unavailable) if the status code *isn't* "Failed", and false when it is?